vent/controller/control_module.py

```diff
@@ -319,148 +234,98 @@ def __analyze_last_waveform(self):
-        # Returns only the inactive alarms
-        self._lock.acquire()
-        logged_alarms = self.COPY_logged_alarms.copy()
-        self._lock.release()
-        return logged_alarms
-
+    def get_alarms(self):
+        """
+        Returns alarms, by time of occurance:
+        """
+        with self._lock:
+            hapa = self.HAPA
+            techa = self.TECHA
 
+        if (hapa is not None) or (techa is not NONE):
+            print("An alarm has happened")
+            #TODO: Link to Alarm manager
+        else:
+            return None
 
     def set_control(self, control_setting: ControlSetting):
         ''' Updates the entry of COPY contained in the control settings'''
-        self._lock.acquire()
-
-        if control_setting.name == ValueName.PIP:
-            self.COPY_SET_PIP = control_setting.value
-            self.COPY_PIP_min = control_setting.min_value
-            self.COPY_PIP_max = control_setting.max_value
-            self.COPY_PIP_lastset = control_setting.timestamp
-
-        elif control_setting.name == ValueName.PIP_TIME:
-            self.COPY_SET_PIP_TIME = control_setting.value
-            self.COPY_PIP_time_min = control_setting.min_value
-            self.COPY_PIP_time_max = control_setting.max_value
-            self.COPY_PIP_time_lastset = control_setting.timestamp
-
-        elif control_setting.name == ValueName.PEEP:
-            self.COPY_SET_PEEP = control_setting.value
-            self.COPY_PEEP_min = control_setting.min_value
-            self.COPY_PEEP_max = control_setting.max_value
-            self.COPY_PEEP_lastset = control_setting.timestamp
-
-        elif control_setting.name == ValueName.BREATHS_PER_MINUTE:
-            self.COPY_SET_BPM = control_setting.value
-            self.COPY_bpm_min = control_setting.min_value
-            self.COPY_bpm_max = control_setting.max_value
-            self.COPY_bpm_lastset = control_setting.timestamp
-
-        elif control_setting.name == ValueName.INSPIRATION_TIME_SEC:
-            self.COPY_SET_I_PHASE = control_setting.value
-            self.COPY_I_phase_min = control_setting.min_value
-            self.COPY_I_phase_max = control_setting.max_value
-            self.COPY_I_phase_lastset = control_setting.timestamp
-
-        elif control_setting.name == ValueName.PEEP_TIME:
-            self.COPY_SET_PEEP_TIME = control_setting.value
-            self.COPY_PEEP_min = control_setting.min_value
-            self.COPY_PEEP_max = control_setting.max_value
-            self.COPY_PEEP_lastset = control_setting.timestamp
 
-        else:
-            raise KeyError("You cannot set the variabe: " + str(control_setting.name))
+        try:  #Make sure the controls are valid  
+            v_is_valid  = control_setting.value > 0
+            n_is_valid = np.sum([k == control_setting.name for k, v in CONTROL.items()]) == 1
+            controls_ok = v_is_valid and n_is_valid
+        except:
+            controls_ok = False
```

Comment:
@gdoubleyew Have added stuff like this whenever the controller interacts with other components of the program. Is this what you had in mind?

Reply:
moving checks like these to the ControlSetting class